src/image/TransmissionCurve.cc

```diff
@@ -0,0 +1,827 @@
+        if (_impl.isSpatiallyConstant) {
+            return shared_from_this();
+        } else {
+            return nullptr;
+        }
+    }
+
+    std::string getPersistenceName() const override { return Impl::NAME; }
+
+    struct PersistenceHelper {
+        table::Schema schema;
+        table::Key<double> throughputAtMin;
+        table::Key<double> throughputAtMax;
+        ArrayKeyVector arrays;
+
+        static PersistenceHelper const & get() {
+            static PersistenceHelper const instance;
+            return instance;
+        }
+
+    private:
+
+        PersistenceHelper() :
+            schema(),
+            throughputAtMin(schema.addField<double>(
+                "throughputAtMin", "throughput below minimum wavelength")),
+            throughputAtMax(schema.addField<double>(
+                "throughputAtMax", "throughput above minimum wavelength"))
+        {
+            Impl::setupPersistence(schema, arrays);
+            schema.getCitizen().markPersistent();
+        }
+
+    };
+
+    void write(OutputArchiveHandle& handle) const override {
+        auto const & keys = PersistenceHelper::get();
+        auto catalog = handle.makeCatalog(keys.schema);
+        auto record = catalog.addNew();
+        record->set(keys.throughputAtMin, _atBounds.first);
+        record->set(keys.throughputAtMax, _atBounds.second);
+        _impl.persist(*record, keys.arrays);
+        handle.saveCatalog(catalog);
+    }
+
+    class Factory : public table::io::PersistableFactory {
+    public:
+
+        std::shared_ptr<table::io::Persistable> read(
+            InputArchive const& archive,
+            CatalogVector const& catalogs
+        ) const override {
+            auto const& keys = PersistenceHelper::get();
+            LSST_ARCHIVE_ASSERT(catalogs.size() == 1u);
+            LSST_ARCHIVE_ASSERT(catalogs.front().getSchema() == keys.schema);
+            auto & record = catalogs.front().front();
+            return std::make_shared<InterpolatedTransmissionCurve>(
+                Impl::unpersist(record, keys.arrays),
+                std::make_pair(record.get(keys.throughputAtMin), record.get(keys.throughputAtMax))
+            );
+        }
+
+        Factory() : table::io::PersistableFactory(Impl::NAME) {}
+    };
+
+    static Factory registration;
+
+private:
+    std::pair<double,double> _atBounds;
+    Impl _impl;
+};
+
+template <typename Impl>
+typename InterpolatedTransmissionCurve<Impl>::Factory InterpolatedTransmissionCurve<Impl>::registration;
+
+template class InterpolatedTransmissionCurve<Impl1d>;
+template class InterpolatedTransmissionCurve<Impl2d>;
+
+
+//-----------------------------------------------------------------------------
+//
+// ProductTransmissionCurve: default for TransmissionCurve::multiply().
+//
+// This is a straightforward lazy-evaluation object.  Its only state is the
+// two operands it delegates to.
+//
+//-----------------------------------------------------------------------------
```

Comment:
Why not just standardize to Doxygen format? (I know you refuse to use actual Doxygen comments for private code, but at least the comment layout could be the same.)